CondFormats/SiPixelObjects/src/PixelROC.cc

```diff
@@ -3,38 +3,282 @@
+//       <<" "<<theFrameConverter.row().slope()
+//       <<" "<<theFrameConverter.collumn().offset()
+//       <<" "<<theFrameConverter.collumn().slope()
+//       <<endl;
+// }
+
+// works for phase 1, find det side from the local method
+void PixelROC::initFrameConversionPhase1() {
+  const bool PRINT = false;
+
+  if(PRINT) cout<<"PixelROC::initFramConversion - for phase1 "<<endl;
+
+  int side = 0;
+  bool isBarrel = PixelModuleName::isBarrel(theDetUnit);
+  if(isBarrel) {
+    side = bpixSidePhase1(theDetUnit); // find the side for phase1
+    if(PRINT) cout<<" bpix, side "<<side<<endl;
   } else {
-    PixelEndcapName endcapName(theDetUnit);
-    theFrameConverter =  FrameConversion(endcapName, theIdDU); 
+    side = fpixSidePhase1(theDetUnit);
+    if(PRINT) cout<<" fpix, side "<<side<<endl;
   }
+
+  theFrameConverter = FrameConversion(isBarrel,side, theIdDU);
+
+  if(PRINT) cout<<" init FrameConversion "
+      <<" "<<theDetUnit<<" "<<theIdDU<<" "<<theIdLk
+      <<" "<<theFrameConverter.row().offset()
+      <<" "<<theFrameConverter.row().slope()
+      <<" "<<theFrameConverter.collumn().offset()
+      <<" "<<theFrameConverter.collumn().slope()
+      <<endl;
 }
 
-string PixelROC::print(int depth) const
-{
+// Works only for phase0, uses the fixed pixel id
+void PixelROC::initFrameConversion() {
+
+    if ( PixelModuleName::isBarrel(theDetUnit) ) {
+      PixelBarrelName barrelName(theDetUnit);
+      //cout<<" bpix mod "<<barrelName.name()<<" "<<theDetUnit<<" "<<theIdDU<<" "<<theIdLk<<endl;
+      theFrameConverter = FrameConversion(barrelName, theIdDU);
+    } else {
+      PixelEndcapName endcapName(theDetUnit);
+      theFrameConverter =  FrameConversion(endcapName, theIdDU); 
+    }
+    // } // end phase 
+
+  // cout<<" init FrameConversion "
+  //     <<" "<<theDetUnit<<" "<<theIdDU<<" "<<theIdLk
+  //     <<" "<<theFrameConverter.row().offset()
+  //     <<" "<<theFrameConverter.row().slope()
+  //     <<" "<<theFrameConverter.collumn().offset()
+  //     <<" "<<theFrameConverter.collumn().slope()
+  //     <<endl;
+
+}
+
+// These are methods to find the module side.
+// The are hardwired for phase0 and phase1
+// Will not work for phase2 or when the detid coding changes.
+int PixelROC::bpixSidePhase0(uint32_t rawId) const {
+  int side = 1;
+  /// two bits would be enough, but  we could use the number "0" as a wildcard
+  //const unsigned int layerStartBit_=   16;
+  //const unsigned int ladderStartBit_=   8;
+  const unsigned int moduleStartBit_=   2;
+  /// two bits would be enough, but  we could use the number "0" as a wildcard
+  //const unsigned int layerMask_=       0xF;
+  //const unsigned int ladderMask_=      0xFF;
+  const unsigned int moduleMask_=      0x3F;
+
+  /// layer id
+  //unsigned int layer = (rawId>>layerStartBit_) & layerMask_;
+  /// ladder  id
+  //unsigned int ladder = (rawId>>ladderStartBit_) & ladderMask_;
+  /// det id
+  unsigned int module = (rawId>>moduleStartBit_)& moduleMask_;
```

Comment:
I have spend some time trying to use the TrackerTopology here. 
I have managed to pass it from the constructor of the raw2digi code.
The code was declared non-tread safe and I had to replace it.
For the moment this is the only way that I know which works.
D.

> On 7 Apr 2016, at 15:52, Vincenzo Innocente <notifications@github.com> wrote:
> 
> In CondFormats/SiPixelObjects/src/PixelROC.cc:
> 
> > +  int side = 1;
> > +  /// two bits would be enough, but  we could use the number "0" as a wildcard
> > +  //const unsigned int layerStartBit_=   16;
> > +  //const unsigned int ladderStartBit_=   8;
> > +  const unsigned int moduleStartBit_=   2;
> > +  /// two bits would be enough, but  we could use the number "0" as a wildcard
> > +  //const unsigned int layerMask_=       0xF;
> > +  //const unsigned int ladderMask_=      0xFF;
> > +  const unsigned int moduleMask_=      0x3F;
> > +
> > +  /// layer id
> > +  //unsigned int layer = (rawId>>layerStartBit_) & layerMask_;
> > +  /// ladder  id
> > +  //unsigned int ladder = (rawId>>ladderStartBit_) & ladderMask_;
> > +  /// det id
> > +  unsigned int module = (rawId>>moduleStartBit_)& moduleMask_;
> 
> because is not accessible from within here!
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
>